### ISSUE TITLE
CompletableFuture with timeout hangs Isolate

### ISSUE THREAD

**Issue by alexfedorenchik** [oracle/graal#4024]
**Describe the issue**
java.util.concurrent.CompletableFuture$Delayer class that is used in "orTimeout" and "completeOnTimeout" methods prevents isolate from being teared-down.

As far as I understand, Delayer instantiates a static ScheduledThreadPoolExecutor with a single daemon thread that is not terminated due to https://github.com/oracle/graal/issues/2617#issuecomment-673015433

Issue can be workarounded by shutting down that executor using reflection, but perhaps it makes sense to provide some generic substitution?

**Steps to reproduce the issue**
Example source (Test.java)
```java
import java.lang.reflect.Method;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.TimeUnit;

import org.graalvm.nativeimage.IsolateThread;
import org.graalvm.nativeimage.Isolates;
import org.graalvm.nativeimage.Isolates.CreateIsolateParameters;
import org.graalvm.nativeimage.c.function.CEntryPoint;

public class Test {
    public static void main(String[] args) {
        IsolateThread ctx = Isolates.createIsolate(CreateIsolateParameters.getDefault());
        System.out.println("Isolate created");
        run(ctx);
        System.out.println("Isolate completed");
        Isolates.tearDownIsolate(ctx);
        System.out.println("Isolate cleared");
    }

    @CEntryPoint
    public static void run(@CEntryPoint.IsolateThreadContext IsolateThread ctx) {
        var exec = Executors.newFixedThreadPool(1);
        CompletableFuture
                .supplyAsync(() -> "Some work is done", exec)
                .whenComplete((result, throwable) -> System.out.println(result))
                .orTimeout(1, TimeUnit.SECONDS)
                .join();
        exec.shutdownNow();
        //Uncomment next line for applying workaround
        //shutdownNowDelayScheduler();
    }

    private static void shutdownNowDelayScheduler() {
        try {
            var delayerClass = Class.forName("java.util.concurrent.CompletableFuture$Delayer");
            var delayerField = delayerClass.getDeclaredField("delayer");
            delayerField.setAccessible(true);
            var delayerValue = delayerField.get(null);
            Method method = ExecutorService.class.getDeclaredMethod("shutdownNow", new Class[]{});
            method.invoke(delayerValue);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
```

Build:
```
echo "Main-Class: Test" > manifest.txt && javac Test.java &&  jar -cvfm Test.jar manifest.txt Test.class && native-image -jar Test.jar
```

Run native-image: 
```
./Test
```

Expected Result:

```
Isolate created
Some work is done
Isolate completed
Isolate cleared
```
app finished

Actual Result:

```
Isolate created
Some work is done
Isolate completed
```
app hangs...

**Describe GraalVM and your environment:**
 - GraalVM version CE 21.2.0
 - JDK major version: 11
 - OS: macOS Monterey 12.0.1
 - Architecture: x86

**Comment by oubidar-Abderrahim** [comment]
Thank you for reporting this, we will take a look into it and get back to you.

**Comment by peter-hofer** [comment]
Fixed in 52b709f17b7384f0997fd76476437d9106652de4.
